### PR TITLE
Add SorbetRails.configure

### PR DESCRIPTION
**Pull request by connorshea**
Add the ability to configure what gem plugins are enabled.

Now you can do this in `config/initializers/sorbet_rails.rb`:

```ruby
SorbetRails.configure do |config|
  config.enabled_gem_plugins = [
    :kaminari
  ]
end
```

And it'll enable the plugin in your Rails app.

There are probably more elegant ways of doing this, but I wasn't able to get it working as just `SorbetRails.enabled_gem_plugins = []`. This is pretty much stolen from Kaminari's own `config` method :D https://github.com/kaminari/kaminari/blob/9e9ac3b7516b409a4aed1ce1bc6c5e90108c9511/kaminari-core/lib/kaminari/config.rb

### CODE REVIEW

File: lib/sorbet-rails/gem_plugins/gem_plugins.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# typed: true
+require('sorbet-rails/config')
+
+module SorbetRails::GemPlugins
+  extend T::Sig
+
+  sig { void }
+  def self.register_gem_plugins
+    if SorbetRails.config.enabled_gem_plugins.include?(:kaminari)
```
Comment (connorshea) [inline lib/sorbet-rails/gem_plugins/gem_plugins.rb]: I should probably add a check here for whether the configured `enable_gem_plugins` includes only valid symbols, and error if there are any invalid symbols. Probably a better UX than doing nothing and never telling the user they've screwed up.

File: lib/sorbet-rails/config.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# typed: true
```
Comment (manhhung741) [inline lib/sorbet-rails/config.rb]: strict?

File: lib/sorbet-rails/gem_plugins/gem_plugins.rb (hunk 1 above)
Comment (manhhung741) [reply lib/sorbet-rails/gem_plugins/gem_plugins.rb]: we can do a switch:
```
SorbetRails.config.enabled_gem_plugins.each do |plugin_sym|
  case plugin_sym
  when :kaminari
     ...
  when :pg_search
    ...
  else
    InvalidPlugin.new
end
```

**Comment by manhhung741** [comment]
I pushed some changes to consolidate the plugin configuration & also register plugins immediately after they are configured. Let me know what you think & feel free to make changes!

File: lib/sorbet-rails/model_plugins/plugins.rb [hunk 3]
```diff
@@ -42,4 +30,43 @@ def set_plugins(plugins)
...
+
+  sig { params(plugin_name: Symbol).void }
+  def register_plugin_by_name(plugin_name)
+    register_plugin(get_plugin_by_name(plugin_name))
+  end
+
+  sig { params(plugin_name: Symbol).returns(T.class_of(Base)) }
+  def get_plugin_by_name(plugin_name)
+    case plugin_name
+    when :active_record_enum
```
Comment (connorshea) [inline lib/sorbet-rails/model_plugins/plugins.rb]: is the intention to allow disabling the core plugins?

**Comment by manhhung741** [comment]
Yea, it’s intentional to be able to remove core plugins. I imagine in case
there is a conflict in a core plugin, people can remove it or subclass it
to deal with their edge case. This way people are not blocked when they
file issue/open PR for improvement.

**Comment by connorshea** [comment]
Hmm, why is this failing with `uninitialized constant T` 🤔

**Pull request merged**